cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraBulkWriterContext.java

```diff
@@ -87,11 +87,21 @@ private CassandraBulkWriterContext(@NotNull BulkSparkConf conf,
         Set<String> udts = CqlUtils.extractUdts(keyspaceSchema, keyspace);
         ReplicationFactor replicationFactor = CqlUtils.extractReplicationFactor(keyspaceSchema, keyspace);
         int indexCount = CqlUtils.extractIndexCount(keyspaceSchema, keyspace, table);
-        CqlTable cqlTable = bridge.buildSchema(createTableSchema, keyspace, replicationFactor, partitioner, udts, null, indexCount);
+        CqlTable cqlTable = bridge().buildSchema(createTableSchema, keyspace, replicationFactor, partitioner, udts, null, indexCount);
 
         TableInfoProvider tableInfoProvider = new CqlTableInfoProvider(createTableSchema, cqlTable);
         TableSchema tableSchema = initializeTableSchema(conf, dfSchema, tableInfoProvider, lowestCassandraVersion);
-        schemaInfo = new CassandraSchemaInfo(tableSchema);
+        schemaInfo = new CassandraSchemaInfo(tableSchema, udts, cqlTable);
+    }
+
+    @Override
+    public synchronized CassandraBridge bridge()
+    {
+        if (this.bridge == null)
+        {
+            this.bridge = CassandraBridgeFactory.get(this.clusterInfo.getLowestCassandraVersion());
```

Comment:
Same comment as above - it's possible the "lowest version" has changed from the beginning of the job. Again, maybe not worth the extra load but I don't think it's particularly heavy-weight.

Reply:
On a smaller cluster maybe not, but I think it can add up on a big Cassandra cluster when running a large spark job. I think there's a tradeoff here. In the case where that one instance coming back with a lower version, the instance would probably miss the write (as it would be unable to import the SStables) and the data would need to be repaired. So the question is whether we want so save some REST endpoint requests on all the job runs vs save on repair cost on a very unlikely scenario that would very rarely happen.

Reply:
It is safe (or safer) to use the low version among the entire job runtime, which is essential the low version when the job starts. 
Cassandra can take the sstable written with low version. 
There could be a risk that a node has to downgrade and the written sstables are in the new version. The scenario is rare, but you get my point.